src/taproot.rs

```diff
@@ -168,13 +177,42 @@ pub mod test_helpers {
 
         (nonces, shares)
     }
+
+    /// Run a tweaked signing round for the passed `msg`
+    #[allow(non_snake_case)]
+    pub fn sign_tweaked<RNG: RngCore + CryptoRng, Signer: traits::Signer>(
+        msg: &[u8],
+        signers: &mut [Signer],
+        rng: &mut RNG,
+        tweaked_public_key: &Point,
+    ) -> (Vec<PublicNonce>, Vec<SignatureShare>) {
+        let signer_ids: Vec<u32> = signers.iter().map(|s| s.get_id()).collect();
+        let key_ids: Vec<u32> = signers.iter().flat_map(|s| s.get_key_ids()).collect();
+        let mut nonces: Vec<PublicNonce> =
+            signers.iter_mut().flat_map(|s| s.gen_nonces(rng)).collect();
+
+        loop {
+            let (_, R) = Signer::compute_intermediate(msg, &signer_ids, &key_ids, &nonces);
+            if R.has_even_y() {
+                break;
+            }
+            nonces = signers.iter_mut().flat_map(|s| s.gen_nonces(rng)).collect();
+        }
+
```

Comment:
Nit: From what I can see, this is the same code as in the `.sign(..)` method and could be factored out to a shared helper function for increased code reuse and readability.

Reply:
Yeah, I did a better job factoring `SignatureAggregator::{sign, sign_merkle_root, sign_with_tweak}` for code reuse.  I'll do something similar for `Party::{sign, sign_tweaked}`.